q2_sample_classifier/classify.py

```diff
@@ -100,6 +100,19 @@ def regress_samples(output_dir: str, table: pd.DataFrame,
                optimize_feature_selection, title='regression predictions')
 
 
+def split_table(table: pd.DataFrame, metadata: qiime2.MetadataColumn,
+                test_size: float=defaults['test_size'], random_state: int=None,
+                stratify: str=True) -> (pd.DataFrame, pd.DataFrame):
+    column = metadata.to_series().name
+    X_train, X_test, y_train, y_test = _prepare_training_data(
+        table, metadata, column, test_size, random_state, load_data=True,
+        stratify=True)
+    # TODO: we can consider returning the metadata (y_train, y_test) if a
+    # SampleData[Metadata] type comes into existence. For now we will just
+    # throw this out.
+    return X_train, X_test
```

Comment:
I'm a little confused here -- are these the biom tables?? Is it a good idea to return them as data frames?
I would think it would be a better idea if these were biom tables instead, just incase the datasets are huge.

I have an example script here on how to do this: https://github.com/mortonjt/diffrank/blob/master/scripts/process.py#L23

Reply:
And having the metadata returned is also strongly advantageous so that coordination between metadata / biom tables are minimized.  Asking all downstream method to perform this preprocessing by themselves already gets quite hairy.  Especially if there are samples that aren't shared between the metadata / biom tables.

Reply:
> I'm a little confused here -- are these the biom tables?? Is it a good idea to return them as data frames?

These are being operated on as dataframes, since scikit-learn (which performs the random sampling) does not handle biom tables.

They will be saved as biom tables within the QZA, so no worries about file size.

> And having the metadata returned is also strongly advantageous so that coordination between metadata / biom tables are minimized. Asking all downstream method to perform this preprocessing by themselves already gets quite hairy. Especially if there are samples that aren't shared between the metadata / biom tables.

I agree that is ideal, but QIIME 2 does not currently have a good way of outputting metadata. This is sort of a temporary solution, but in truth we will want to have downstream steps perform validation and re-sorting _anyway_ to make sure that the correct metadata files are used as input and are in the same order. This results in fewer files for users to keep track of: they just hold onto their original metadata file, which will be a superset of any tables they have.

Reply:
This is a good point — but a much larger issue.

I have raised this as a separate issue (#99). biom tables are not supported elsewhere — because they are not supported by scikit-learn — so this will require much more extensive changes (probably using a SparseDataFrame or other sparse array), out of scope of this PR.

Supporting biom in split_table will be relatively straightforward, but downstream methods still require a DataFrame (where sparsity will matter more), so for the purposes of q2-sample-classifier it is pointless to fix the one without fixing others. More general fixes to the architecture will be required.